lib/components/viewers/RouteRow.js

```diff
@@ -68,6 +106,31 @@ const SimpleRouteRenderer = styled.span`
   padding: 0;
 `
 
+const PatternButton = styled.button`
+  background: transparent;
+  border: none;
+  border-radius: 50%;
+  color: #004f96;
+  opacity: ${(props) => (props.display ? 100 : 0)};
+  height: 40px;
+  margin-right: 8px;
+  position: relative;
+  transition: all ease-in-out 0.2s;
+  width: 40px;
+  z-index: ${(props) => (props.display ? 2 : -2)};
+
+  svg {
+    margin-top: -3px;
+    height: 22px !important;
+    width: 22px !important;
+  }
+
+  &:hover {
+    background: #eee;
+    transition: all ease-in-out 0.3s;
```

Comment:
Can we speed this up? The delay is annoying when I deselect a row and have to wait for the button to disappear. Also, could we change the function to `ease-out`? I worry that the delayed feedback gives the perception of lag